tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -939,3 +963,152 @@ def _get_num_gpus_on_instance(instance_type_info):
+    """Update cluster to include pre-install script, which introduce bootstrap error."""
+    updated_config_file = pcluster_config_reader(config_file="pcluster.config.update.ini", bucket=bucket_name)
+    cluster.config_file = str(updated_config_file)
+    _update_and_start_cluster(cluster)
+
+
+def _set_protected_failure_count(remote_command_executor, test_datadir, protected_failure_count):
+    """Disable protected mode by setting protected_failure_count to -1."""
+    remote_command_executor.run_remote_command(
+        f"echo 'protected_failure_count = {protected_failure_count}' | sudo tee -a "
+        "/etc/parallelcluster/slurm_plugin/parallelcluster_clustermgtd.conf"
+    )
+    remote_command_executor.run_remote_script(str(test_datadir / "slurm_restart_clustermgtd.sh"), run_as_root=True)
+
+
+def _enable_protected_mode(remote_command_executor, test_datadir):
+    """Enable protected mode by removing lines related to protected mode in the config, so it will be set to default."""
+    remote_command_executor.run_remote_command(
+        "sudo sed -i '/'protected_failure_count'/d' /etc/parallelcluster/slurm_plugin/parallelcluster_clustermgtd.conf"
+    )
+    remote_command_executor.run_remote_script(str(test_datadir / "slurm_restart_clustermgtd.sh"), run_as_root=True)
+
+
+def _test_disable_protected_mode(remote_command_executor, test_datadir):
+    """Test Bootstrap failures have no affect on cluster when protected mode is disabled."""
+    # Disable protected mode by setting protected failure count to -1
+    _set_protected_failure_count(remote_command_executor, test_datadir, -1)
+    # wait till the node failed
+    retry(wait_fixed=seconds(20), stop_max_delay=minutes(15))(assert_errors_in_logs)(
+        remote_command_executor,
+        ["/var/log/parallelcluster/clustermgtd"],
+        [
+            "Found the following unhealthy static nodes",
+        ],
+    )
+    # Assert that it does not contain bootstrap failure
+    assert_no_msg_in_logs(
+        remote_command_executor,
+        ["/var/log/parallelcluster/clustermgtd"],
+        ["Node bootstrap error"],
+    )
+    # Re-enable protected mode
+    _enable_protected_mode(remote_command_executor, test_datadir)
+
+
+def _test_active_job_running(scheduler_commands, remote_command_executor):
+    """Test cluster is not placed into protected mode when there is an active job running even reach threshold."""
+    # Submit a job to the queue contains broken nodes and normal node, submit the job to the normal node to test
+    # the queue will not be disabled if there's active job running.
+    cancel_job_id = scheduler_commands.submit_command_and_assert_job_accepted(
+        submit_command_args={"command": "sleep 3000", "nodes": 1, "partition": "half-broken", "constraint": "c5.xlarge"}
+    )
+    # Sleep for a bit for the job to run, also for some iteration of clustermgtd to change node states
+    time.sleep(120)
```

Comment:
Do we need to do this? I would imagine the problematic job would take a bit of time to fail anyway, so maybe we can just get rid of this sleep time?

Reply:
If two jobs submit together, the normal node maybe not ready to start before the pre-install bootstrap failure happen, have been reduce it to 15, fail, now reduced the sleep time here to 60.

Reply:
Sounds like we should assert that the normal node is running the job before submitting the job with bootstrap failure to enforce the sequence of events, instead of just using some arbitrary sleep time here.

Reply:
Added a function to to wait for job running